app/models/customer.rb

```diff
@@ -0,0 +1,21 @@
+class Customer < ApplicationRecord
+
+  has_many :rentals, dependent: :nullify
+  validates :name, presence: true
+
+  def self.check_out_movie(id)
+    customer = Customer.find_by(id: id)
+    movie_count = customer.movies_checked_out_count
+    customer.update(movies_checked_out_count: movie_count + 1)
+
+    customer.save
+    return customer
+  end
```

Comment:
So the `checked_out_count` has nothing to do with the number of `Rentals` the customer has.  Hmmm... Any kind of subtle bug could get these out of sync.  This isn't an out-and-out bug, but in an imperfect system it's an invitation for these kinds of problems.